contracts/reverseRegistrar/IL2ReverseRegistrar.sol

```diff
@@ -0,0 +1,24 @@
+pragma solidity >=0.8.4;
+
+interface IL2ReverseRegistrar {
+    function setName(string memory name) external returns (bytes32);
+
+    function setNameForAddrWithSignature(
+        address addr,
+        string memory name,
+        address relayer,
+        uint256 signatureExpiry,
```

Comment:
I wonder if, rather than using expiry, we should use inception? That gives us an ordering over signatures and we know that we can ignore earlier ones if we have a later one.

Reply:
What do you mean by inception? Do you mean a nonce-based system?